### PR TITLE
FEATURE: BB-1406 basic subtotals support

### PR DESCRIPTION
**Pull request by yanlipnican**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3155/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2502/ : FAILURE in 1m 18s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3156/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2503/ : FAILURE in 1m 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3157/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2504/ : FAILURE in 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3158/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2505/ : FAILURE in 1m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3159/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2506/ : SUCCESS in 5m 59s

File: src/helpers/mappingHeader.ts [hunk 1]
```diff
@@ -58,4 +59,7 @@ export function getMappingHeaderUri(header: IMappingHeader): string {
     if (isMappingHeaderMeasureItem(header)) {
         return header.measureHeaderItem.uri;
     }
+    if (isMappingHeaderTotal(header)) {
+        return header.totalHeaderItem.name;
```
Comment (martinmosko) [inline src/helpers/mappingHeader.ts]: return name instead of expected uri `getMappingHeaderUri`

File: src/helpers/agGrid.ts [hunk 2]
```diff
@@ -235,31 +235,63 @@ export const getFields = (dataHeaders: Execution.IResultHeaderItem[][]) => {
...
+
+    if (Execution.isTotalHeaderItem(rowHeaderDataItem)) {
+        const totalName = rowHeaderDataItem.totalHeaderItem.name;
+        return {
+            ...cell,
+            value: getSubtotalLabelCellIndex(rowHeaderData, rowIndex) === rowHeaderIndex
+                ? intl.formatMessage({ id: `visualizations.totals.dropdown.title.${totalName}` })
+                : null
+        };
+    }
```
Comment (martinmosko) [inline src/helpers/agGrid.ts]: missing invariant to exclude `isMeasureHeaderItem`

File: src/helpers/agGrid.ts [hunk 3]
```diff
@@ -235,31 +235,63 @@ export const getFields = (dataHeaders: Execution.IResultHeaderItem[][]) => {
...
+    return resultHeaderItems.findIndex(headerItem => Execution.isTotalHeaderItem(headerItem[rowIndex]));
+};
+
+const getCell = (
+    rowHeaderData: Execution.IResultHeaderItem[][],
+    rowIndex: number,
+    rowHeader: IGridHeader,
+    rowHeaderIndex: number,
+    intl: InjectedIntl
+) => {
```
Comment (martinmosko) [inline src/helpers/agGrid.ts]: return type

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3160/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2507/ : SUCCESS in 6m 22s

File: src/helpers/agGrid.ts (hunk 2 above)
Comment (yanlipnican) [reply src/helpers/agGrid.ts]: Done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3161/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2508/ : SUCCESS in 2m 38s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3200/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2541/ : FAILURE in 12m 00s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3204/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2545/ : FAILURE in 9m 49s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3206/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2546/ : FAILURE in 13m 10s

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3227/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2558/ : FAILURE in 9m 36s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by yanlipnican** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3229/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2560/ : SUCCESS in 10m 31s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1343/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/3230/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/806/ : UNSTABLE in 23m 57s

**Comment by petrjanu** [comment]
LTGM

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/689/ : FAILURE in 28s
- gooddata-react-components-unit-tests-zuul-docker-el7 : SKIPPED
- gooddata-react-components-storybook-zuul-docker : SKIPPED
- gooddata-react-components-examples-zuul-docker : SKIPPED

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/690/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2562/ : SUCCESS in 4m 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1344/ : UNSTABLE in 52m 36s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/673/ : SUCCESS in 26m 19s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/691/ : SUCCESS in 38s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/2563/ : SUCCESS in 4m 20s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1345/ : SUCCESS in 28m 31s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/674/ : SUCCESS in 21m 45s

**Pull request merged**